services/worker/src/worker/job_runners/split/image_url_columns.py

```diff
@@ -0,0 +1,82 @@
+# SPDX-License-Identifier: Apache-2.0
+# Copyright 2023 The HuggingFace Authors.
+
+import logging
+
+from libcommon.constants import PROCESSING_STEP_SPLIT_IMAGE_URL_COLUMNS_VERSION
+from libcommon.exceptions import PreviousStepFormatError
+from libcommon.utils import is_image_url
+
+from worker.job_runners.split.split_job_runner import SplitJobRunner
+from worker.utils import (
+    CompleteJobResult,
+    ImageUrlColumnsResponse,
+    SplitFirstRowsResponse,
+    get_previous_step_or_raise,
+)
+
+
+def compute_image_url_columns(
+    dataset: str,
+    config: str,
+    split: str,
+) -> ImageUrlColumnsResponse:
+    logging.info(f"get image-url-columns for dataset={dataset} config={config} split={split}")
+
+    # get the first rows from previous job
+    upstream_response = get_previous_step_or_raise(
+        kinds=["split-first-rows-from-streaming", "split-first-rows-from-parquet"],
+        dataset=dataset,
+        config=config,
+        split=split,
+    )
+    try:
+        first_rows_response = upstream_response.response
+        upstream_response_content = SplitFirstRowsResponse(
+            dataset=dataset,
+            config=config,
+            split=split,
+            features=first_rows_response["content"]["features"],
+            rows=first_rows_response["content"]["rows"],
+        )
+
+        features = upstream_response_content["features"]
+        first_rows = upstream_response_content["rows"]
+    except KeyError as e:
+        raise PreviousStepFormatError("Previous step did not return the expected content.", e) from e
+
+    # look for URLs columns using the first rows
+    string_type_dict = {"dtype": "string", "_type": "Value"}
+    string_columns = [feature["name"] for feature in features if feature["type"] == string_type_dict]
```

Comment:
Possibly `feature["type"]` can contain extra fields, so I think it would be safer to check the existence and value of `dtype` and `_type`, than checking the equality between objects

Reply:
Done